hw3/01-routing.js

```diff
@@ -1,51 +1,97 @@
+// http://localhost:5000/cookie should return 'cookies… yummm' in plain text and set 'hello=world' as a cookie
 
-// http://localhost:5001/check-cookies should return 'yes' / 'no' in plain text depending on whether the browser has the 'hello' cookie
+// http://localhost:5000/check-cookies should return 'yes' / 'no' in plain text depending on whether the browser has the 'hello' cookie
 
-// For other routes, such as http://localhost:5001/other, this exercise should return a status code 404 with '404 - page not found' in html format
+// For other routes, such as http://localhost:5000/other, this exercise should return a status code 404 with '404 - page not found' in html format
 
 const server = http.createServer((req, res) => {
   const routes = [
-    'welcome',
-    'redirect',
-    'redirected',
-    'cache',
-    'cookie',
-    'check-cookies',
-    'other',
+    "home",
+    "welcome",
+    "redirect",
+    "redirected",
+    "cache",
+    "cookie",
+    "check-cookies",
+    "other",
   ];
 
   let getRoutes = () => {
-    let result = '';
-
+    let result = "";
     routes.forEach(
       (elem) => (result += `<li><a href="/${elem}">${elem}</a></li>`)
     );
-
     return result;
   };
 
-  if (req.url === '/') {
+  if (req.url === "/" || req.url === "/home") {
     let routeResults = getRoutes();
-
-    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.writeHead(200, { "Content-Type": "text/html" });
     res.write(`<h1>Exercise 01</h1>`);
     res.write(`<ul> ${routeResults} </ul>`);
     res.end();
   }
 
   // Add your code here
+  else if (req.url === "/welcome") {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write(`<h2> WELCOME! </h2>`);
+    res.write(`<a href="/home"> Back to home </a>`);
+    res.end();
+  } else if (req.url === "/redirect") {
+    let redirectLink = "http://localhost:5000/redirected";
+    res.writeHead(302, { Location: redirectLink });
+    res.end();
+  } else if (req.url === "/redirected") {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write(`<h2> Redirected! </h2>`);
+    res.write(`<a href="/home"> Back to home </a>`);
+    res.end();
+  } else if (req.url === "/cache") {
+    res.writeHead(200, {
+      "Content-Type": "text/html",
+      "Cache-Control": "max-age=86400",
+    });
+    res.write(`<h2> This resource was cached </h2>`);
+    res.write(`<a href="/home"> Back to home </a>`);
+    res.end();
+  } else if (req.url === "/cookie") {
+    res.writeHead(200, {
+      "Content-Type": "text/html",
+      "Set-Cookie": "hello=world",
+    });
+    res.write(`<h2> cookies... yummm </h2>`);
+    res.write(`<a href="/home"> Back to home </a>`);
+    res.end();
+  } else if (req.url === "/check-cookies") {
+    Cookie = req.headers.cookie;
+    console.log("Print the cookie");
+    console.log(Cookie);
+    res.writeHead(200, { "Content-Type": "text/html" });
+    if (Cookie === "hello=world") {
```

Comment:
Hello Aashna,

I think it would be a good idea to iterate over all the cookies to check if has the 'hello world' cookie, as there can be more than one cookie set in a session.